src/native/managed/cdacreader/src/Legacy/ISOSDacInterface.cs

```diff
@@ -442,9 +442,165 @@ internal unsafe partial interface ISOSDacInterface2
     int IsRCWDCOMProxy(ulong rcwAddress, int* inDCOMProxy);
 }
 
+[GeneratedComInterface]
+[Guid("B08C5CDC-FD8A-49C5-AB38-5FEEF35235B4")]
+internal unsafe partial interface ISOSDacInterface3
+{
+    [PreserveSig]
+    int GetGCInterestingInfoData(ulong interestingInfoAddr, /*struct DacpGCInterestingInfoData*/ void* data);
+    [PreserveSig]
+    int GetGCInterestingInfoStaticData(/*struct DacpGCInterestingInfoData*/ void* data);
+    [PreserveSig]
+    int GetGCGlobalMechanisms(nuint* globalMechanisms);
+};
+
+[GeneratedComInterface]
+[Guid("74B9D34C-A612-4B07-93DD-5462178FCE11")]
+internal unsafe partial interface ISOSDacInterface4
+{
+    [PreserveSig]
+    int GetClrNotification([In, Out, MarshalUsing(CountElementName = nameof(count))] ulong[] arguments, int count, int* pNeeded);
```

Comment:
Can't this just be marked as `ref ulong[]` instead of with the `[In, Out]` attribute?

Reply:
Oh right, this is the array value issue. `ref ulong[]` would result in `ulong**` on the native side.